credoai/modules/dataset_modules/dataset_base.py

```diff
@@ -58,101 +70,170 @@ def prepare_results(self):
             )
     
     def _group_differences(self):
-        group_means = self.X.groupby(self.sensitive_features).mean()
-        std = self.X.std(numeric_only=True)
+        """Calculates standardized mean differences
+        It is performed for all numeric features and all possible group pairs combinations present in the sensitive feature.
+
+        Returns
+        -------
+        dict, nested
+            Key: senstive feature groups pair
+            Values: dict
+                Key: name of feature
+                Value: standardized mean difference
+        """
+        X = self.data.drop(columns=[self.sensitive_feature_key, self.label_key])
+        sensitive_features = self.data[self.sensitive_feature_key]
+
+        group_means = X.groupby(sensitive_features).mean()
+        std = X.std(numeric_only=True)
         diffs = {}
         for group1, group2 in combinations(group_means.index, 2):
             diff = (group_means.loc[group1]-group_means.loc[group2])/std
             diffs[f'{group1}-{group2}'] = diff.to_dict()
 
         return diffs
     
-    def _run_cv(self, pipe):
+    def _run_cv(self):
+        """Determines cross-validated ROC-AUC score
+        A model is trained on the features to predict the sensitive attribute.
+        The score quantifies the performance of this prediction. 
+        A high score means the data collectively serves as a proxy.
+
+        Parameters
+        ----------
+        pipe : sklearn.pipeline
+            Pipeline of transforms
+
+        Returns
+        -------
+        ndarray
+            Cross-validation score
+        """
+        results = {}
+        pipe = self._make_pipe()
+        X = self.data.drop(columns=[self.sensitive_feature_key, self.label_key])
+        sensitive_features = self.data[self.sensitive_feature_key].cat.codes
```

Comment:
Using `cat.codes` which gets out numerical representations of the sensitive feature. Necessary for xgboost.